clippy_lints/src/dereference.rs

```diff
@@ -1503,12 +1512,25 @@ fn report<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, state: State, data
                 "&"
             };
 
-            let expr_str = if !expr_is_macro_call && is_final_ufcs && expr.precedence().order() < PREC_PREFIX {
+            let mut expr_str = if !expr_is_macro_call && is_final_ufcs && expr.precedence().order() < PREC_PREFIX {
                 format!("({expr_str})")
             } else {
                 expr_str.into_owned()
             };
 
+            // Fix #10850, changes suggestion if it's `Foo::deref` instead of `foo.deref`. Since `Foo::deref` is
+            // a `Call` instead of a `MethodCall` this should catch all instances of this, even if it's fully
+            // qualified or whatnot.
+            if is_final_ufcs && let Some(args) = call_args {
+                // Remove ref if it's there
+                let arg = if let ExprKind::AddrOf(.., arg) = args[0].kind {
```

Comment:
Is there a function that does this for me? Since this does feel a tad hacky.

Reply:
I also feel this could break on something like `*foo` if it's `&&Foo` but I haven't tested this yet.